### PR TITLE
[llvm] Replace the LLVM got array with a set of scalar variables whic…  …h are initialized by a callback called by the runtime.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -2079,8 +2079,11 @@ init_amodule_got (MonoAotModule *amodule, gboolean preinit)
 			amodule->got [i] = amodule->shared_got [i];
 	}
 	if (amodule->llvm_got) {
-		for (i = 0; i < npatches; ++i)
+		void (*init_aotconst) (int, gpointer) = (void (*)(int, gpointer))amodule->info.llvm_init_aotconst;
```
Comment (jaykrell) [inline mono/mini/aot-runtime.c]: typedef?

File: mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -4704,6 +4710,11 @@ init_method (MonoAotModule *amodule, guint32 method_index, MonoMethod *method, M
 				got [got_slots [pindex]] = addr;
 				if (ji->type == MONO_PATCH_INFO_METHOD_JUMP)
 					register_jump_target_got_slot (domain, ji->data.method, &(got [got_slots [pindex]]));
+
+				if (llvm) {
+					void (*init_aotconst) (int, gpointer) = (void (*)(int, gpointer))amodule->info.llvm_init_aotconst;
```
Comment (jaykrell) [inline mono/mini/aot-runtime.c]: typedef?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
So this works now, but leads to some code size increase on wasm, i.e. the generated init_aotconst () functions can get quite big (+67k on an aot hello world). An alternative would be putting the variable addresses in an array, but that might inhibit optimization.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**